File: template-project-microservice/template-project-frontend/src/services/user-service.js
/**
 * 用户服务
 */

import userApi from '@api/user.js';
import logger from '@utils/logger.js';

class UserService {
  /**
   * 获取用户列表
   */
  async getUsers(params = {}) {
    try {
      const response = await userApi.getUsers(params);
      // 处理分页响应
      if (response.data && response.data.list) {
        return {
          list: response.data.list,
          total: response.data.total || 0,
          pageNum: response.data.pageNum || 1,
          pageSize: response.data.pageSize || 10,
          pages: response.data.pages || 0
        };
      }
      // 处理简单列表响应
      return {
        list: response.data || [],
        total: response.data?.length || 0,
        pageNum: 1,
        pageSize: 10,
        pages: 1
      };
    } catch (error) {
      logger.error('Failed to get users', error);
      throw error;
    }
  }

  /**
   * 获取用户详情
   */
  async getUserById(id) {
    try {
      const response = await userApi.getUserById(id);
      return response.data || response;
    } catch (error) {
      logger.error('Failed to get user', error);
      throw error;
    }
  }

  /**
   * 创建用户
   */
  async createUser(data) {
    try {
      const response = await userApi.createUser(data);
      logger.info('User created', { id: response.data?.id });
      return response.data || response;
    } catch (error) {
      logger.error('Failed to create user', error);
      throw error;
    }
  }

  /**
   * 更新用户
   */
  async updateUser(id, data) {
    try {
      const response = await userApi.updateUser(id, data);
      logger.info('User updated', { id });
      return response.data || response;
    } catch (error) {
      logger.error('Failed to update user', error);
      throw error;
    }
  }

  /**
   * 删除用户
   */
  async deleteUser(id) {
    try {
      await userApi.deleteUser(id);
      logger.info('User deleted', { id });
    } catch (error) {
      logger.error('Failed to delete user', error);
      throw error;
    }
  }

  /**
   * 批量删除用户
   */
  async batchDeleteUsers(ids) {
    try {
      await userApi.batchDeleteUsers(ids);
      logger.info('Users batch deleted', { count: ids.length });
    } catch (error) {
      logger.error('Failed to batch delete users', error);
      throw error;
    }
  }
}

export default new UserService();

File: template-project-microservice/template-project-cloud/cloud-modules/user-service/src/main/java/com/example/user/application/service/UserService.java
package com.example.user.application.service;

import com.example.core.common.exception.BusinessException;
import com.example.core.common.result.ResultCode;
import com.example.user.application.dto.CreateUserCommand;
import com.example.user.application.dto.PageResult;
import com.example.user.application.dto.UpdateUserCommand;
import com.example.user.application.dto.UserDTO;
import com.example.user.domain.entity.UserInfo;
import com.example.user.domain.repository.UserInfoRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.util.StringUtils;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

/**
 * 用户服务
 */
@Service
@RequiredArgsConstructor
public class UserService {
    private final UserInfoRepository userInfoRepository;

    /**
     * 获取用户列表（分页）
     */
    public PageResult<UserDTO> getUserList(int pageNum, int pageSize, String keyword) {
        List<UserInfo> users = userInfoRepository.findList(pageNum, pageSize, keyword);
        long total = userInfoRepository.count(keyword);

        List<UserDTO> userDTOs = users.stream()
                .map(this::convertToDTO)
                .collect(Collectors.toList());

        return new PageResult<>(userDTOs, total, pageNum, pageSize);
    }

    /**
     * 根据ID获取用户
     */
    public UserDTO getUserById(String id) {
        UserInfo user = userInfoRepository.findById(id);
        if (user == null) {
            throw new BusinessException(ResultCode.NOT_FOUND, "用户不存在");
        }
        return convertToDTO(user);
    }

    /**
     * 创建用户
     */
    @Transactional
    public UserDTO createUser(CreateUserCommand command) {
        // 检查用户名是否已存在
        UserInfo existingUser = userInfoRepository.findByUsername(command.getUsername());
        if (existingUser != null) {
            throw new BusinessException(ResultCode.PARAM_ERROR, "用户名已存在");
        }

        UserInfo user = new UserInfo();
        user.setUsername(command.getUsername());
        user.setEmail(command.getEmail());
        user.setPhone(command.getPhone());
        user.setCreateDate(LocalDateTime.now());
        user.setCreateUser("system");
        user.setDeleted(0);
        user.setDbVersion(1);

        userInfoRepository.save(user);
        return convertToDTO(user);
    }

    /**
     * 更新用户
     */
    @Transactional
    public UserDTO updateUser(String id, UpdateUserCommand command) {
        UserInfo user = userInfoRepository.findById(id);
        if (user == null) {
            throw new BusinessException(ResultCode.NOT_FOUND, "用户不存在");
        }

        if (StringUtils.hasText(command.getEmail())) {
            user.setEmail(command.getEmail());
        }
        if (StringUtils.hasText(command.getPhone())) {
            user.setPhone(command.getPhone());
        }
        if (command.getStatus() != null) {
            // 注意：这里假设UserInfo有status字段，如果没有需要从auth_user表关联查询
            // 暂时先不处理status
        }

        user.setUpdateDate(LocalDateTime.now());
        user.setUpdateUser("system");
        user.setDbVersion(user.getDbVersion() + 1);

        userInfoRepository.update(user);
        return convertToDTO(user);
    }

    /**
     * 删除用户
     */
    @Transactional
    public void deleteUser(String id) {
        UserInfo user = userInfoRepository.findById(id);
        if (user == null) {
            throw new BusinessException(ResultCode.NOT_FOUND, "用户不存在");
        }
        userInfoRepository.delete(id);
    }

    /**
     * 转换为DTO
     */
    private UserDTO convertToDTO(UserInfo user) {
        UserDTO dto = new UserDTO();
        dto.setId(user.getId());
        dto.setUsername(user.getUsername());
        dto.setNickname(user.getNickname());
        dto.setEmail(user.getEmail());
        dto.setPhone(user.getPhone());
        dto.setAvatar(user.getAvatar());
        dto.setGender(user.getGender());
        dto.setBirthday(user.getBirthday());
        dto.setAddress(user.getAddress());
        dto.setCreateDate(user.getCreateDate());
        dto.setUpdateDate(user.getUpdateDate());
        // status需要从auth_user表查询，这里暂时设为1
        dto.setStatus(1);
        return dto;
    }
}



